src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/Program.cs

```diff
@@ -79,136 +235,381 @@ private static void DoTheRun(int iterations)
+        }
 
-            // Pool both channel factories and channels, recycle both factories and channels while using them 
-            RecyclablePooledFactoriesAndChannels_OpenOnce<WcfService1.IService1, HelloWorldTest>.RunAllScenariosWithWeights(100, 1, 1);
-            //RecyclablePooledFactoriesAndChannels_OpenOnce<WcfService1.IDuplexService, DuplexTest>.RunAllScenariosWithWeights(100, 1, 1);
-            RecyclablePooledFactoriesAndChannels_OpenOnce<WcfService1.IStreamingService, StreamingTest>.RunAllScenariosWithWeights(100, 1, 1);
-            
+        public static void RunFirstNIterations(int iterations, int maxTasks, Action test)
+        {
+            RunFirstNIterationsAsyncImpl(iterations, maxTasks, () => { test(); return Task.FromResult(true); });
+        }
+        public static void RunFirstNIterationsAsync(int iterations, int maxTasks, Func<Task> testAsync)
+        {
+            RunFirstNIterationsAsyncImpl(iterations, maxTasks, testAsync);
+        }
+        public static void RunFirstNIterationsAsyncImpl(int iterations, int maxTasks, Func<Task> testAsync)
+        {
+            Task[] allTasks = new Task[maxTasks];
 
-            // Pool both factories and channels, use their async methods, recycle both factories and channels while using them
-            RecyclablePooledFactoriesAndChannelsAsync_OpenOnce<WcfService1.IService1, HelloWorldTest>.RunAllScenariosWithWeightsAsync(1000, 1, 1).Wait();
-            //RecyclablePooledFactoriesAndChannelsAsync_OpenOnce<WcfService1.IDuplexService, DuplexTest>.RunAllScenariosWithWeightsAsync(1000, 1, 1).Wait();
-            //RecyclablePooledFactoriesAndChannelsAsync_OpenOnce<WcfService1.IStreamingService, StreamingTest>.RunAllScenariosWithWeightsAsync(1000, 1, 1).Wait();
+            Console.WriteLine("Start RunFirstNIterations with " + iterations + " iterations");
+            var sw = new Stopwatch();
+            sw.Start();
 
-            // Add more async tests to achieve parity with sync tests
+            for (int t = 0; t < allTasks.Length; t++)
+            {
+                int tt = t;
+                allTasks[t] = Task.Run(async () =>
+                {
+                    var ttt = tt;
+                    for (int i = 0; i < iterations / allTasks.Length; i++)
+                    {
+                        try
+                        {
+                            await testAsync();
+                        }
+                        catch (ObjectDisposedException e)
+                        {
+                            TestUtils.ReportFailure(e.ToString());
+                        }
+                        catch (Exception e)
+                        {
+                            TestUtils.ReportFailure(e.ToString());
+                            throw;
+                        }
+#if DEBUG
+                        if (i % 100 == 0)
+                        {
+                            Console.WriteLine(DateTime.Now.ToString() + " " + ttt + " " + i);
```

Comment:
Just a question: I notice Console is used other places not inside an 'if DEBUG'.  Should they all be inside 'if DEBUG' checks?